build_product_images.py

```diff
@@ -149,8 +185,55 @@ def product_to_build(product_name, product_version, products):
         return None
 
 
+def check_platform(architecture):
+    """
+    Checks if a desired platform is given, gives current platform if not
+    """
+    if architecture is None or len(architecture) == 0:
+        architecture = ["linux/" + platform.machine()]
```

Comment:
I don't know, I tested now loosely what would happen if I give docker a --platform linux/X86_64 and it looks like it still fetches an amd64. I tested it with tools and with java-base 1.8.0, inspecting those images suggest that it is amd64.

If you think this might end in odd behaviour or errors which could be avoided I am still happy to delete that too.